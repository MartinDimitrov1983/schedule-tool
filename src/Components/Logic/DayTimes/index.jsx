import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { changeTime } from '../../../Redux/slices/days';
import { TimePicker } from '../..';

const DayTimes = ({
  isTemplate,
  isCompleted,
  dayIndex,
  timeIndex,
  removeHandler,
  autocompleteData,
}) => {
  const times = useSelector((state) => state.days.times);
  const dispatch = useDispatch();

  const onChange = (value) => {
    dispatch(changeTime({ dayIndex, timeIndex, value }));
  };

  return (
    <TimePicker
      isTemplate={isTemplate}
      isCompleted={isCompleted}
      onChange={onChange}
      value={
        autocompleteData ? autocompleteData : times?.[dayIndex]?.[timeIndex]
      }
      onRemoveClick={removeHandler}
    />
  );
};

export default DayTimes;
