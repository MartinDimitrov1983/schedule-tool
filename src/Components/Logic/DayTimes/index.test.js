import React from 'react';
import { render } from '@testing-library/react';

import DayTimes from './index';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

test('Snapshot DayTimes', () => {
  const { asFragment } = render(
    <DayTimes
      isTemplate={false}
      isCompleted={false}
      dayIndex={0}
      timeIndex={0}
      removeHandler={() => {}}
      autocompleteData={''}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
