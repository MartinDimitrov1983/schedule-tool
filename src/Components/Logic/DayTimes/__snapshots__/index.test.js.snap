// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot DayTimes 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="react-time-picker react-time-picker--closed react-time-picker--enabled autoComplete wrapper"
    >
      <div
        class="react-time-picker__wrapper"
      >
        <div
          class="react-time-picker__inputGroup"
        >
          <input
            hidden=""
            name="time"
            step="60"
            style="visibility: hidden; position: absolute; z-index: -999;"
            type="time"
            value=""
          />
          <input
            autocomplete="off"
            class="react-time-picker__inputGroup__input react-time-picker__inputGroup__hour"
            data-input="true"
            inputmode="numeric"
            max="23"
            min="0"
            name="hour24"
            placeholder="--"
            type="number"
            value=""
          />
          <span
            class="react-time-picker__inputGroup__divider"
          >
            :
          </span>
          <input
            autocomplete="off"
            class="react-time-picker__inputGroup__input react-time-picker__inputGroup__minute"
            data-input="true"
            inputmode="numeric"
            max="59"
            min="0"
            name="minute"
            placeholder="--"
            type="number"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
